hydra-node/test/Hydra/API/ServerSpec.hs

```diff
@@ -58,6 +58,37 @@ spec = parallel $ do
               failAfter 1 $ atomically (replicateM 2 (readTQueue queue)) `shouldReturn` [arbitraryMsg, arbitraryMsg]
               failAfter 1 $ atomically (tryReadTQueue queue) `shouldReturn` Nothing
 
+  it "sends all sendOutput history to all connected clients after a restart" $ do
+    queue1 <- atomically newTQueue
+    queue2 <- atomically newTQueue
+    showLogsOnFailure $ \tracer -> failAfter 5 $
+      withTempDir "ServerSpec" $ \tmpDir -> do
+        let persistentFile = tmpDir <> "/history"
+        let arbitraryMsg = ReadyToCommit mempty
+
+        persistence <- createPersistenceIncremental persistentFile
+        withFreePort $ \port -> do
+          withAPIServer @SimpleTx "127.0.0.1" (fromIntegral port) alice persistence tracer noop $ \Server{sendOutput} -> do
+            sendOutput arbitraryMsg
+
+        persistence' <- createPersistenceIncremental persistentFile
```

Comment:
Don't need to re-initialize persistence. We can re-use the same handle as above.

Reply:
I wanted to re-initialize persistence here to be closer to the actual code. When the node restarts, it will reinitialize persistence before starting the server. I'm confident that it has no impact but... I don't know, maybe some shit could happen during the persistent initialization if it evolves in the future.